src/parser/postgresparser.cpp

```diff
@@ -454,12 +452,12 @@ std::unique_ptr<AbstractExpression> PostgresParser::AExprTransform(A_Expr *root)
 }
 
 // Postgres.BoolExpr -> terrier.ConjunctionExpression
-std::unique_ptr<AbstractExpression> PostgresParser::BoolExprTransform(BoolExpr *root) {
+std::unique_ptr<AbstractExpression> PostgresParser::BoolExprTransform(ParseResult *parse_result, BoolExpr *root) {
   std::unique_ptr<AbstractExpression> result;
-  std::vector<std::shared_ptr<AbstractExpression>> children;
+  std::vector<std::unique_ptr<AbstractExpression>> children;
```

Comment:
This should be a vector of managed pointers. When you move children in and children is a vector of unique_ptrs are the unique_ptrs maintained? When children is cleaned up at the end of this function would the unique_ptr destructor be called or is children empty at this point because all of its contents would have been moved?

Reply:
Nvm I think it should be fine based on move semantics